clustersConfig.py

```diff
@@ -233,7 +233,7 @@ def __init__(
             self.masters.append(NodeConfig(self.name, **n))
 
         self.configured_workers = [NodeConfig(self.name, **w) for w in cc["workers"]]
-        self.workers = [NodeConfig(self.name, **w) for w in worker_range.filter(cc["workers"])]
+        self.workers = list(worker_range.filter(self.configured_workers))
```

Comment:
This makes the config depend on the args. I think we should give each worker an ID and then use that ID to pass into a function that returns a mac for that node instead while we still create all workers.

Reply:
Giving it id means that we need to generate config with id, or track the id as part of this function both seem to be a bit clunky

Reply:
Currently, we have:

```
    mac: str = field(default_factory=lambda: mac_generator.next_mac())
```

That's pretty close to having an ID as well except that the ID is in the mac. I'm wondering if the bug you found is due to

```
self.workers = [NodeConfig(self.name, **w) for w in worker_range.filter(cc["workers"])]
```
Which makes the config depend on the args. Can we store the range somewhere else? We need a `CdaConfig` above ClustersConfig that contains things like `secrets_path` and `worker_range`. Whenever, you need the range of workers you are configuring, you would call a function where you pass in the range as well. That way the config is more immutable too.

Would that solve the issue?

Reply:
That would probably solve that, the unfortunate thing is that there is at least 40 places where would need to adjust the call to workers.

Reply:
The reason that the hash of the ip would work is because you are specifying a unique ip manually?

Reply:
Yeah, it's already in the config and it has to be unique

Reply:
It's already in **your** config, iiuc?

Reply:
It is, but with this question I suppose you are not using statically configured IPs right?